browser_utils.js

```diff
@@ -557,6 +557,34 @@ async function clickSelector(page, selector, {timeout=getDefaultTimeout(page), c
     }
 }
 
+/**
+ * Asserts that a selector is not present in the passed page or frame.
+ *
+ * @example
+ * ```javascript
+ * await assertNotSelector(page, 'div[data-id="foo"] a.view', {message: 'Expected foo to not be present'});
+ * ```
+ * @param {import('puppeteer').Page} page puppeteer page object.
+ * @param {string} selector [CSS selector](https://www.w3.org/TR/2018/REC-selectors-3-20181106/#selectors) (aka query selector) of the targeted element.
+ * @param {{timeout?: number, message?: string}} [__namedParameters] Options (currently not visible in output due to typedoc bug)
+ * @param {string?} message Error message shown if the element is not visible in time.
+ * @param {number?} timeout How long to wait, in milliseconds.
+ */
+async function assertNotSelector(page, selector, {timeout=getDefaultTimeout(page), message} = {}) {
+    const config = getBrowser(page)._pentf_config;
+    addBreadcrumb(config, `enter assertNotSelector(${selector})`);
+    try {
+        await page.waitForSelector(selector, {timeout});
+        throw new Error(`Element matching ${selector} is present, but should not be there. ${message ? ' ' + message : ''}`);
+    } catch(err) {
+        if (/Element\smatching/.test(err.message)) {
```

Comment:
(disregard prior comment)
Mmm, this seems a bit like a hack. Couldn't we `return` here and throw below?